homeassistant/components/plex/sensor.py

```diff
@@ -103,6 +132,109 @@ def device_info(self):
             "identifiers": {(PLEX_DOMAIN, self._server.machine_identifier)},
             "manufacturer": "Plex",
             "model": "Plex Media Server",
-            "name": "Activity Sensor",
+            "name": self._server.friendly_name,
+            "sw_version": self._server.version,
+        }
+
+
+class PlexLibrarySectionSensor(SensorEntity):
+    """Representation of a Plex library section sensor."""
+
+    def __init__(self, hass, plex_server, plex_library_section):
+        """Initialize the sensor."""
+        self._server = plex_server
+        self.server_name = plex_server.friendly_name
+        self.server_id = plex_server.machine_identifier
+        self.library_section = plex_library_section
+        self.library_type = plex_library_section.type
+        self._name = f"{self.server_name} Library - {plex_library_section.title}"
+        self._unique_id = f"library-{self.server_id}-{plex_library_section.uuid}"
+        self._state = None
+        self._attributes = {}
+
+    async def async_added_to_hass(self):
+        """Run when about to be added to hass."""
+        unsub = async_dispatcher_connect(
+            self.hass,
+            PLEX_UPDATE_LIBRARY_SIGNAL.format(self.server_id),
+            self.async_refresh_sensor,
+        )
```

Comment:
```suggestion
        unsub = async_dispatcher_connect(
            self.hass,
            PLEX_UPDATE_LIBRARY_SIGNAL.format(self.server_id),
            self.async_refresh_sensor,
        )
```

Can you wrap this in `self.async_on_remove` instead?